src/routes/provider-edit-route/provider-edit-route.test.js

```diff
@@ -0,0 +1,269 @@
+    <Harness>
+      <ProviderEditRoute
+        getProvider={mockGetProvider}
+        getProxyTypes={mockGetProxyTypes}
+        getRootProxy={mockGetRootProxy}
+        history={history}
+        location={location}
+        match={match}
+        model={model}
+        proxyTypes={proxyTypes}
+        removeUpdateRequests={mockRemoveUpdateRequests}
+        rootProxy={rootProxy}
+        updateProvider={mockUpdateProvider}
+        {...props}
+      />
+        Page content
+    </Harness>
+  </MemoryRouter>
+);
+
+const renderProviderEditRoute = (props) => render(getProviderEditRoute(props));
+describe('Given ProviderEditRoute', () => {
+  beforeEach(() => {
+    mockGetProxyTypes.mockClear();
+    mockGetRootProxy.mockClear();
+    mockRemoveUpdateRequests.mockClear();
+    mockUpdateProvider.mockClear();
+    history.replace.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should render page', () => {
+    const { getByText } = renderProviderEditRoute({});
+
+    expect(getByText('Page content')).toBeDefined();
+  });
+
+  it('should handle getProvider with providerId', async () => {
+    await act(async () => {
+      await renderProviderEditRoute();
+    });
+
+    expect(mockGetProvider).toHaveBeenCalledWith('provider-id');
+  });
+
+  it('should handle getProvider', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: {
+          getProvider: mockGetProvider,
+          match: {
+            ...match,
+            params: { providerId: 'other-provider-id' },
+          },
+        },
+      });
+    });
+
+    expect(mockGetProvider).toHaveBeenCalled();
+  });
+
+  it('should handle GetProxyTypes', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: { proxyType : mockGetProxyTypes },
+      });
+    });
+
+    expect(mockGetProxyTypes).toHaveBeenCalled();
+  });
+
+  it('should handle GetProxyTypes', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: { proxyType : mockGetProxyTypes },
+      });
+    });
+
+    expect(mockGetProxyTypes).toHaveBeenCalled();
+  });
+
+  it('should handle Cancel', async () => {
+    let getByRole;
+
+    await act(async () => {
+      getByRole = await renderProviderEditRoute({}).getByRole;
+    });
+
+    fireEvent.click(getByRole('button', { name: 'ui-eholdings.label.icon.closeX' }));
+
+    expect(history.replace).toHaveBeenCalledWith({
+      pathname: `/eholdings/providers/${model.id}`,
+      search: location.search,
+      state: {
+        eholdings: true,
+      },
+    });
+  });
+
+  it('should handle onSubmit', async () => {
+    let getByRole;
+
+    await act(async () => {
+      getByRole = await renderProviderEditRoute().getByRole;
+    });
+
+    fireEvent.submit(getByRole('button', { name: 'stripes-components.saveAndClose' }));
+
+    expect(mockUpdateProvider).toHaveBeenCalled();
+  });
+  describe('when update request resolves', () => {
```

Comment:
Please add empty lines between `describe`

Reply:
Thanks, I fixed it.